cmd/config.go

```diff
@@ -106,21 +113,32 @@ func newConfigGetCmd(stack *string) *cobra.Command {
 				return errors.Wrap(err, "invalid configuration key")
 			}
 
-			return getConfig(s, key, jsonOut)
+			return getConfig(s, key, path, jsonOut)
 		}),
 	}
 	getCmd.Flags().BoolVarP(
 		&jsonOut, "json", "j", false,
 		"Emit output as JSON")
+	getCmd.PersistentFlags().BoolVar(
+		&path, "path", false,
+		"The key contains a path to a property in a map or list to get")
 
 	return getCmd
 }
 
 func newConfigRmCmd(stack *string) *cobra.Command {
+	var path bool
+
 	rmCmd := &cobra.Command{
 		Use:   "rm <key>",
 		Short: "Remove configuration value",
-		Args:  cmdutil.SpecificArgs([]string{"key"}),
+		Long: "Remove configuration value.\n\n" +
+			"The `--path` flag can be used to remove a value inside a map or list:\n\n" +
+			"    - `pulumi config rm --path outer.inner` will remove the `inner` key, " +
```

Comment:
I didn't see any integration tests for some of the more exotic ways you can try to reference configuration.

Do scenarios like the following work? Do we expect them to?
```
# Setting a config value in some deeply nested thing that
# doesn't already exist?
pulumi config set semiInner 1 --path "a.b.c.d"
pulumi config set wayInner 1 --path "a.b.c.d.e.f.g.h.i.j.k"

# Removing some nested config value that doesn't exist.
# We should return an error, but confirm we don't, say, panic.
pulumi config rm foo --path "non-exist.a.b.c.d.e.f.g.h.i.j.k"

# What about some horrible combination of array and object dereferences?
pulumi config set x '{"a": [ { "b": [ { "c": [ { "d": [ { "e": 1 } ] } ] } ] } ] }'
pulumi config get x[0].a[0].c[0].d[0].e
```

Reply:
I'll add integration tests that actually call `pulumi config`.

Reply:
BTW, `--path` is not a string, it's a bool flag that indicates the key _contains_ a path.

So your examples would be more like:

```
# Setting a config value in some deeply nested thing that
# doesn't already exist?
pulumi config set --path semiInner.a.b.c.d 1
pulumi config set --path wayInner.a.b.c.d.e.f.g.h.i.j.k 1
```

> Removing some nested config value that doesn't exist.
> We should return an error, but confirm we don't, say, panic.

Today, running `pulumi config rm doesnotexist` does not result in an error, so I made it behave the same for paths inside an object that don't exist: no error for `rm`.